packages/next/src/utils/generate-globs.ts

```diff
@@ -1,27 +1,76 @@
-import { joinPathFragments } from '@nrwl/devkit';
+import { joinPathFragments, logger } from '@nrwl/devkit';
 import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';
 import { appRootPath } from '@nrwl/workspace/src/utilities/app-root';
-import { getSourceDirOfDependentProjects } from '@nrwl/workspace/src/utilities/project-graph-utils';
-import { resolve } from 'path';
+import {
+  getProjectNameFromDirPath,
+  getSourceDirOfDependentProjects,
+} from '@nrwl/workspace/src/utilities/project-graph-utils';
+import { relative, resolve } from 'path';
 
 /**
- * Creates an array of of the dependent projects' source directories, appended
- * with the passed glob pattern.
- * @param projectName workspace relative filename, you can pass `__filename` if executed in a node process
- * @param fileGlobPattern the glob pattern to be applied to the various project directories
- * @returns
+ * generates a set of glob patterns based off the source root of the app and it's dependencies
+ * @param dirPath workspace relative directory path that will be used to infer the parent project and dependencies
+ * @param fileGlobPattern pass a custom glob pattern to be used
+ */
+export function createGlobPatternsForDependencies(
+  dirPath: string,
+  fileGlobPattern: string = '/**/!(*.stories|*.spec).{tsx,jsx}'
```

Comment:
🤔 do we need to include `*.jsx` files as well? Same question for the React glob export

//cc @jaysoo @JamesHenry ?

Reply:
Hmm yeah tricky - I never have, but I believe people do, and of course Nx does support that use case via the generators and glob patterns in tsconfigs etc

I don't know whether a boolean to additionally include JS (false by default) is worth it over and above just requiring the user to provide a custom glob. In either case the user would need to understand the context...

The alternative of just always including JS is not ideal either as it would slow things down for folks not using that approach (of course questionable how much this inefficiency actually matters in the real world).

@juristr On that point maybe the next step should be for us to understand what operations tailwind will perform on files which match the glob pattern and try and assess how expensive they are? If very cheap we can err on the side of ease of use and not worry about this

Reply:
> I don't know whether a boolean to additionally include JS (false by default) is worth it over and above just requiring the user to provide a custom glob.

yeah I wouldn't add a boolean, rather have the user customize the glob. But I guess that could be easily missed. I mean...we would have docs around this, but still 🤔.

And yes, including all JS could be an overhead and include a lot of files which are not needed.

> On that point maybe the next step should be for us to understand what operations tailwind will perform on files which match the glob pattern behind the scenes and try and assess how expensive they are?

yep, we could look into that...though it might be that they're "cheap" today and with coming versions get more expensive I guess. Because if we include `tsx, jsx, js` then sooner or later we can also just include all the files 😅

Reply:
People use `.js` over `.jsx` actually.  I haven't seen much `.jsx` usage in the wild.

Reply:
@jaysoo ok, so you'd suggest using a pattern like `tsx,js` instead?